source/MRMesh/MRDistanceMap.cpp

```diff
@@ -531,20 +531,17 @@ Polyline2 distanceMapTo2DIsoPolyline( const DistanceMap& distMap, float isoValue
     };
     auto horizontalEdgesSize = ( resX - 1 ) * resY;
     std::vector<SeparationPoint> separationPoints( horizontalEdgesSize + resX * ( resY - 1 ) );
+    tbb::enumerable_thread_specific<size_t> numValidVertsPerThread( 0 );
     auto setupSeparation = [&] ( size_t x0, size_t y0, size_t x1, size_t y1 )
     {
-        if ( x1 == resX || y1 == resY )
-            return;
-        const auto v0opt = distMap.get( x0, y0 );
-        const auto v1opt = distMap.get( x1, y1 );
-        if ( !v0opt || !v1opt )
-            return;
-        const auto v0 = *v0opt;
-        const auto v1 = *v1opt;
+        const auto v0 = distMap.getValue( x0, y0 );
+        const auto v1 = distMap.getValue( x1, y1 );
+        if ( v0 == NOT_VALID_VALUE || v1 == NOT_VALID_VALUE )
+            return false;
```

Comment:
please return decimal type or use explicit conversion when adding result of this function to counter